index.js

```diff
@@ -87,31 +79,19 @@ module.exports = Metrics = {
     )
   },
 
-  buildPromKey(key) {
-    if (key == null) {
-      key = ''
-    }
+  buildPromKey(key = '') {
     return key.replace(/[^a-zA-Z0-9]/g, '_')
   },
 
   sanitizeValue(value) {
     return parseFloat(value)
   },
 
-  set(key, value, sampleRate) {
-    if (sampleRate == null) {
-      sampleRate = 1
-    }
+  set(key, value, sampleRate = 1) {
```

Comment:
The `sampleRate` parameter is ignored (in all kinds of metrics), so we could remove it param -- or leave it for backwards compatibility, in case we add another param in the future.

Reply:
Good point. I didn't realize that. It comes from the statsd implementation, which we don't have anymore. I'm all for removing that param, but it would require a major version bump, and we might want to take the opportunity to remove other statsd-isms. I'd do that in another PR.

Reply:
Dropping the default value assignment for the`sampleRate` should not require a major version bump -- it's ignored anyways.

In this PR we could keep the signature `set(key, value, sampleRate) {` and _save_ the cpu cycles -- which in my testing for my translations implementation are more than the simple null-check in the body (I was comparing `t(key, vars={}) {...}` vs `t(key, var) { var=vars||{}; ...}`).

Reply:
It's true that sampleRate is ignored by the set() function, but all other functions also [take a 4th opts parameter](https://github.com/overleaf/metrics-module/pull/40/files#diff-168726dbe96b3ce427e7fedce31bb0bcR99). Removing the sampleRate parameter would move the opts parameter to 3rd position.

I don't mind removing the default parameters; they are slightly backwards incompatible (if you explicitly pass null as an argument instead of omitting it). However, I'm not convinced that they're slower. I'd be curious to see your benchmark comparing null checks to default values. I compared the three following functions:
```js
function defaultParams(x = {}) {
  return x;
}

function nullFallback(x) {
  if (x == null) {
    x = {}
  }
  return x
}

function orFallback(x) {
  x = x || {}
  return x
}
```
and obtained no significant difference:
```
default params x 793,799,215 ops/sec ±0.10% (97 runs sampled) Result: {}
null fallback x 794,175,403 ops/sec ±0.08% (97 runs sampled) Result: {}
or fallback x 792,914,074 ops/sec ±0.15% (99 runs sampled) Result: {}
Fastest is null fallback,default params,or fallback
```

Reply:
I was running an earlier revision of this [`perf` test-suite](https://github.com/das7pad/translations-sharelatex/blob/6d4a5c2b5e23a82ae49fdbe1614bcf6e16ff710c/test/unit/js/translations-tests.js#L183), but I cannot recall when I performed the testing. It was prior moving the `translate` function into a build pipeline for sure. Running the tests just now does not show any significant difference either. So I may be wrong here, possibly I saw differences due to CPU throttling that happened to hit some runs more than others :shrug: . Let's ship this as is and drop the parameter altogether in the next major version.